mantidproject/mantid
# Remove duplicate member function Kernel::DataItem::name().

Description of work.

This removes the (C++) member function `Kernel::DataItem::name()` which is identical to `API::Workspace::getName()`, but returns a new `std::string` instead of a const reference.  These changes should remove many unnecessary copies.

I also removed the python member function `Workspace::getName()` which duplicates `DataItem::name()`, isn't commonly used, and isn't terribly pythonic. 

The latter would be even better as a read-only data member ( `DataItem::name` ) , but having both at the same time leads to runtime errors. [This change](https://github.com/mantidproject/mantid/files/680805/dataitem_name.txt) would break many scripts and is most likely best done later as part of "Mantid 4.0".  Alternatively, we could move the python interface to only use `DataItem::getName()` and add a read-only data member ( `DataItem::name` ). 

**To test:**



Review changes to `Kernel::DataItem` and `API::Workspace`. Most other changes are renaming that should be caught by the build servers.

There is no GitHub issue associated with this pull request

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
